packages/core/src/types/authMachine.ts

```diff
@@ -1,37 +1,83 @@
-import { CognitoUser, CognitoUserSession } from 'amazon-cognito-identity-js';
-import { Interpreter, State } from 'xstate';
+import { CognitoUser } from 'amazon-cognito-identity-js';
+import { Actor, Interpreter, State, StateMachine } from 'xstate';
 import { ValidationError } from './validator';
 
 export type AuthFormData = Record<string, string>;
+
 export interface AuthContext {
-  remoteError?: string; // contains Amplify or Cognito error
-  validationError?: ValidationError; // contains validation error for each input
   user?: CognitoUserAmplify;
-  username?: string;
-  session?: CognitoUserSession;
-  formValues?: AuthFormData;
   config?: {
     login_mechanisms: string[];
   };
-  challengeName?: AuthChallengeNames;
+  actorRef?: any;
+}
+
+export interface SignInContext {
+  remoteError?: string;
+  validationError?: ValidationError;
+  formValues?: AuthFormData;
+  user?: CognitoUserAmplify;
+  challengeName?: string;
+  authAttributes?: Record<string, any>;
+  intent?: string;
+}
+
+export interface SignUpContext {
+  remoteError?: string;
+  validationError?: ValidationError;
+  formValues?: AuthFormData;
+  user?: CognitoUserAmplify;
+  login_mechanisms?: string[];
+  intent?: string;
+  authAttributes?: Record<string, any>;
 }
 
-interface CognitoUserAmplify extends CognitoUser {
+export interface ResetPasswordContext {
+  validationError?: ValidationError;
+  remoteError?: string;
+  formValues?: ValidationError;
   username?: string;
 }
+export interface SignOutContext {
+  user?: CognitoUserAmplify;
+}
+
+// actors that have forms. Has `formValues, remoteErrror, and validationError in common.
+export type ActorContextWithForms =
+  | SignInContext
+  | SignUpContext
+  | ResetPasswordContext;
+
+export type SignInState = State<SignInContext, AuthEvent>;
+export type SignUpState = State<SignUpContext, AuthEvent>;
+export type SignOutState = State<SignOutContext, AuthEvent>;
+export type ResetPasswordState = State<ResetPasswordContext, AuthEvent>;
+export type AuthActorContext =
+  | SignInContext
+  | SignUpContext
+  | SignOutContext
+  | ResetPasswordContext;
```

Comment:
```suggestion
export type AuthActorContext = ActorContextWithForms | SignOutContext;
```

Reply:
`ActorContextWithForms` contains the `SignUp/SignIn/ResetPassword` contexts, so we probably don't need to list them twice. Unless you are worried about `ActorContextWithForms` possibly growing and getting out of sync with this `AuthActorContext` (though from the naming, I don't think it will?).

Reply:
(this is a "nit", btw. not blocking if you don't like the idea)

Reply:
No, I like this! Comming it in.